backend/apps/owasp/utils/compliance_detector.py

```
@@ -0,0 +1,80 @@
+        latest_metrics = ProjectHealthMetrics.get_latest_health_metrics().select_related("project")
+        metrics_to_update = []
+        
+        for metric in latest_metrics:
+            project = metric.project
+            project_name = project.name
+            local_level = str(project.level)
+            
+            # Compare official level with local level
+            # Compare official level with local level
+            lookup_name = project_name.strip()
+            if lookup_name in normalized_official_levels:
```

Ruff (0.12.2)

Undefined name `normalized_official_levels`

(F821)